zkvm/src/merkle.rs

```diff
@@ -32,16 +32,32 @@ enum MerkleNode {
 
 impl MerkleTree {
     /// Constructs a new MerkleTree based on the input list of entries.
-    pub fn build<M: MerkleItem>(label: &'static [u8], list: &[M]) -> Option<Self> {
-        if list.len() == 0 {
-            return None;
-        }
-        let t = Transcript::new(label);
-        Some(MerkleTree {
+    pub fn build<M: MerkleItem>(label: &'static [u8], list: &[M]) -> Self {
+        let root = match list.len() {
+            0 => None,
+            _ => {
+                let t = Transcript::new(label);
+                Some(Self::build_tree(t, list))
+            }
+        };
+        MerkleTree {
             size: list.len(),
             label,
-            root: Self::build_tree(t, list),
-        })
+            root: root,
```

Comment:
Can be written simply
```suggestion
            root,
```